### PR TITLE
Header component

### PR DESCRIPTION
**Pull request by vuonga1103**
- Styled using makeStyles()
- Used axios to fetch from mock API

**Desktop View (Logged In)**

- image
- image

**Mobile View (Logged In)**

- image
- image

**Desktop View (Not Logged In)**

- image

**Mobile View (Not Logged In)**

- image
- image

### CODE REVIEW

File: src/Header/Header.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      .catch(err => console.log(err))
+  }, [])
+
+  const getMenuButtons = () => {
+    if (menuHeaders.length) {
+      return menuHeaders.map((item) => {
+        let color = (item.label === "Join Us!") ? '#50E3C2' : 'white';
+
+        return (<Button key={item.id} color="inherit" href={item.href} className={classes.menuButton} style={{color}}>{item.label}</Button>)
+      })
```
Comment (cfernando415) [inline src/Header/Header.js:50]: use es6 attrib declaration, for example:

return (
          <Button
            {...{
              key: id,
              color: "inherit",
              href,
              className: menuButton,
              style: { color },
            }}
          >
            {label}
          </Button>
        );
      });

File: src/Header/Header.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  return (
+    <header className={classes.root}>
+      <AppBar position="static" className={classes.header}>
+        <Toolbar>
+          
+          <Typography variant="h6" className={classes.title}>
+            Femmecubator
+          </Typography>
+          
+          <div className={classes.menuButtonsContainer}>{ menuButtons }</div>
```
Comment (cfernando415) [inline src/Header/Header.js:65]: className={menuButtonsContainer}

**Review by cfernando415 — CHANGES_REQUESTED** [review]
please see comments and we need unit test

File: src/Header/Header.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    size: '24px',
+  },
+}));
+
+export default function Header() {
+  const classes = useStyles();
+  const [menuHeaders, setMenuHeaders] = useState([])
+
+  useEffect(() => {
+    axios.get('/api/commonMenu')
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/Header/Header.js]: completed [9d92cce](https://github.com/cfernando415/femmecubator-ui-repo/pull/10/commits/9d92cce628993192396df88925bb11ecb59144a2)

File: src/Header/Header.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      .then(res => setMenuHeaders(res.data.headers))
+      .catch(err => console.log(err))
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/Header/Header.js:42]: 👍

File: src/Header/Header.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  },
+}));
+
+export default function Header() {
+  const classes = useStyles();
+  const [menuHeaders, setMenuHeaders] = useState([])
+
+  useEffect(() => {
+    axios.get('/api/commonMenu')
+      .then(res => setMenuHeaders(res.data.headers))
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/Header/Header.js]: completed in 3ebd36682461031c0b3aeab8338af42f23e6b0cc

File: src/Header/Header.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      .catch(err => console.log(err))
+  }, [])
+
+  const getMenuButtons = () => {
+    if (menuHeaders.length) {
+      return menuHeaders.map((item) => {
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/Header/Header.js:48]: completed in 3ebd36682461031c0b3aeab8338af42f23e6b0cc

File: src/Header/Header.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    flexGrow: 1,
+    textAlign: 'left',
+    fontFamily: 'Work Sans, sans-serif',
+    fontWeight: 600,
+    size: '24px',
+  },
+}));
+
+export default function Header() {
+  const classes = useStyles();
```
[the remark this answers is not in the document]
Comment (vuonga1103) [reply src/Header/Header.js]: completed in 3ebd36682461031c0b3aeab8338af42f23e6b0cc

File: src/Header/Header.js (hunk 1 above)
Comment (vuonga1103) [reply src/Header/Header.js]: completed in 3ebd36682461031c0b3aeab8338af42f23e6b0cc

File: src/Header/Header.js (hunk 2 above)
Comment (vuonga1103) [reply src/Header/Header.js]: completed in 3ebd36682461031c0b3aeab8338af42f23e6b0cc

File: src/Header/Header.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  title: {
+    flexGrow: 1,
+    textAlign: 'left',
+    fontFamily: 'Work Sans, sans-serif',
+    fontWeight: 600,
+    size: '24px',
+  },
+}));
+
+export default function Header() {
```
[the remark this answers is not in the document]
Comment (cfernando415) [reply src/Header/Header.js]: issue resolved in commit # https://github.com/cfernando415/femmecubator-ui-repo/pull/10/commits/7d4c9beaf38326c67e82faa376d18b211283e9fe 
using useAuth() to check if user is authenticated or not 👍

File: src/components/Header/Header.test.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+describe('<Header />', () => {
+  beforeEach(() => {
+    render(
+      <AuthProvider>
+        <Header />
+      </AuthProvider>
+    );
+
+    cleanup;
```
Comment (cfernando415) [inline src/components/Header/Header.test.js:36]: put this in aftereach as a callback

afterEach(cleanup);

File: src/components/Header/Header.test.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import {
+  cleanup,
+  fireEvent,
```
Comment (cfernando415) [inline src/components/Header/Header.test.js:4]: remove unused imports

File: src/components/Header/Header.js [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+    const setResponsiveness = () => {
+      return window.innerWidth < 799
+        ? setMobileView(true)
+        : setMobileView(false);
+    };
+
+    setResponsiveness();
+
+    window.addEventListener('resize', () => setResponsiveness());
+  }, []);
```
Comment (cfernando415) [inline src/components/Header/Header.js:117]: add auth as a dependency:

}, [auth]);

File: src/components/Header/Header.js [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+  const [userName, setUserName] = useState('');
+  const [mobileView, setMobileView] = useState(false);
+  const [anchorEl, setAnchorEl] = useState(false);
+  const [drawerOpen, setDrawerOpen] = useState(false);
+
+  useEffect(() => {
+    axios
+      .get(API_PATH.COMMON_MENU)
+      .then(({ data }) => {
+        if (data.headers && auth.isLoggedIn()) {
```
Comment (cfernando415) [inline src/components/Header/Header.js:95]: if(data && data.headers && auth.isLoggedIn())

File: src/components/Header/Header.js [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+import Link from '@material-ui/core/Link';
+import { Drawer } from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+  },
+  header: {
+    backgroundColor: '#3E50B4',
+    ['@media (min-width: 799px)']: {
```
Comment (cfernando415) [inline src/components/Header/Header.js:24]: '@media (min-width: 799px)': {

File: src/components/Header/Header.js [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+        if (data.headers && auth.isLoggedIn()) {
+          const { headers, userName } = data;
+          setMenuHeaders(headers);
+          setUserName(userName);
+        } else {
+          setMenuHeaders(DEFAULT_COMMON_MENU.headers);
+        }
+      })
+      .catch((err) => {
+        console.log(err);
```
Comment (cfernando415) [inline src/components/Header/Header.js:104]: remove console.log

File: src/components/Header/Header.js [hunk 15]
```diff
@@ -0,0 +1,282 @@
...
+      >
+        Femmecubator
+      </Link>
+    </Typography>
+  );
+
+  const getMenuButtons = () => {
+    if (menuHeaders.length) {
+      return menuHeaders
+        .filter(({ href }) => href !== '/logout' && href !== '/account')
```
Comment (cfernando415) [inline src/components/Header/Header.js:132]: i don't think this is needed

File: src/components/Header/Header.js [hunk 16]
```diff
@@ -0,0 +1,282 @@
...
+      <Link
+        {...{ href: '/', color: 'inherit', style: { textDecoration: 'none' } }}
+      >
+        Femmecubator
+      </Link>
+    </Typography>
+  );
+
+  const getMenuButtons = () => {
+    if (menuHeaders.length) {
```
Comment (cfernando415) [inline src/components/Header/Header.js:130]: menuHeaders && menuHeaders.length

**Review by cfernando415 — CHANGES_REQUESTED** [review]
please see comments

File: src/components/Header/Header.test.js (hunk 10 above)
Comment (vuonga1103) [reply src/components/Header/Header.test.js]: fixed in 7e6c6003d60350fc9b2eb63ec1aecf1599f3dfdc

File: src/components/Header/Header.test.js (hunk 9 above)
Comment (vuonga1103) [reply src/components/Header/Header.test.js]: fixed in 7e6c6003d60350fc9b2eb63ec1aecf1599f3dfdc

**Review by cfernando415 — APPROVED** [review]
👍

**Pull request merged**